### PR TITLE
Clipboard

### PR DESCRIPTION
**Pull request by drowsy810301**
copy nthucsoj@gmail.com to clipboard

### CODE REVIEW

**Comment by nickelcheng** [comment]
Is it possible to use bower to download ```ZeroClipboard.js``` ?

**Comment by henryyang42** [comment]
https://github.com/zeroclipboard/zeroclipboard

Maybe this would help.

**Comment by drowsy810301** [comment]
fixed,use bower to download

File: func.py [hunk 1]
```diff
@@ -35,11 +36,18 @@ def write_ini_file(host, db, user, pwd):
     ini_file.close()
 
 def write_email_file(user, pwd):
-    email_file = open('emailInfo.py', 'w') 
+    email_file = open('emailInfo.py', 'w')
     email_file.write('EMAIL_HOST_USER = \'%s\'\n' % user)
     email_file.write('EMAIL_HOST_PASSWORD = \'%s\'\n' % pwd)
     email_file.close()
 
+def create_flash_dir():
```
Comment (henryyang42) [inline func.py]: I you are using bower, these are not needed anymore.

File: index/static/js/clipboard.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+var path = "http://" + location.host + "/static/flash/ZeroClipboard.swf";
```
Comment (henryyang42) [inline index/static/js/clipboard.js]: It should map to our local file.

File: install.py [hunk 3]
```diff
@@ -43,6 +43,16 @@
     email_host_pwd = getpass.getpass('Email host\'s password : ')
     write_email_file(email_host, email_host_pwd)
 
+if not os.path.isdir('./index/static/flash/'):
```
Comment (henryyang42) [inline install.py]: If you are using bower, these are not needed anymore.

**Comment by drowsy810301** [comment]
All comments fixed

File: index/static/js/clipboard.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+var path = "static/zeroclipboard/dist/ZeroClipboard.swf";
+ZeroClipboard.config( { swfPath: path } );
+var client = new ZeroClipboard( document.getElementById("contact_us") );
+client.on( "ready", function( readyEvent ) {
+      // alert( "ZeroClipboard SWF is ready!" );
```
Comment (nickelcheng) [inline index/static/js/clipboard.js]: If the comments are useless, please remove them.

**Comment by nickelcheng** [comment]
+1 beside the comment.
@bruce3557 please check it

**Comment by drowsy810301** [comment]
comment fixed

**Comment by drowsy810301** [comment]
conflict fixed

**Comment by drowsy810301** [comment]
please review again,thanks

**Comment by bruce3557** [comment]
+1, please merge latest code and I can merge it

**Comment by drowsy810301** [comment]
latest code merged

**Pull request merged**